Rubberduck.VBEEditor/Events/TypingCodeEventArgs.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor.Events
+{
+    public class AutoCompleteEventArgs : EventArgs
+    {
+        public AutoCompleteEventArgs(ICodePane pane)
+        {
+            CodePane = pane;
+            var selection = pane.Selection;
+            using (var module = pane.CodeModule)
+            {
+                var atSelection = module.GetLines(selection);
+                if (string.IsNullOrWhiteSpace(atSelection))
+                {
+                    IsCommitted = true;
+                    OldCode = module.GetLines(selection.PreviousLine);
+                }
+                else
+                {
+                    OldCode = module.GetLines(selection);
+                }
+            }
+        }
+
+        public ICodePane CodePane { get; }
+
+        public bool IsCommitted { get; }
+
+        public string OldCode { get; }
+
+        public string NewCode { get; set; }
```

Comment:
needs a `<summary>` comment that clarifies that this is the code **at the current caret position**, after the changes are made. This is used by the invoking service to cache the current line of code and determine whether to do something about the caret being hidden (if the code didn't change, does nothing)